src/du/native/clonesize.c

```
@@ -141,25 +299,61 @@ static int intern_group(const char *name) {
+    CacheRec *crecs;                    // cache write-back records
+    size_t   ncrecs, crec_cap;
 } ThreadOut;
 
-static void ext_push(ThreadOut *t, uint64_t dev, uint64_t len, int group, int node) {
+/**
+ * A globally unique id for one scanned file, minted per worker so no atomics are
+ * needed: 6 bits of thread index, 26 bits of sequence. Only files whose extents
+ * we actually collect get one, which is what the outside-sharing detection below
+ * needs — see merge_pass's single-file cluster accounting.
+ */
+static uint32_t next_file_id(ThreadOut *t) {
```

Clang (14.0.6)

[warning] parameter name 't' is too short, expected at least 3 characters

(readability-identifier-length)